### ISSUE TITLE
Preload / Display & Visible all called at the same time in ASCellNode (ASCollectionNode)

### ISSUE THREAD

**Issue by AssyriaGameStudio** [TextureGroup/Texture#1567]
I have an ASCollectionNode that's contained within an ASViewController. It's laid out using layoutSpecThatFits and the root ASDisplayNode is set to automaticallyManagesSubnodes. 

The ASCollectionNode is functionally working as expected with the one issue that unfortunately when overriding didEnterPreloadState / didEnterDisplayState / didEnterVisibleState on the ASCellNode; all are called at exactly the same time (as the first pixel appears on screen). 

I've attempted calling setTuningParameters on the ASCollectionNode manually to force the range it pre-loads without any improvement (again it just calls the preload / display / visible states all at once / the same time - just as it enters the visible screen space).

This is preventing images being decided before they appear on screen and hurting user experience.

I can't see anything I'm doing wrong / have been attempting to debug it for a few days, and now believe it to be a bug. I'm using the latest texture version (2.8.1).

**Comment by mankusti** [comment]
Hi @AssyriaGameStudio,

Have you found any workaround for this issue or learned anything new? I'm facing the same issue and it really is hurting UX. I'm currently using Texture from master branch.

**Comment by AssyriaGameStudio** [comment]
Yeah - so it turned out that using a custom UICollectionViewFlowLayout was the issue (I was applying some parallax effects on my cells using it).

I amended my custom UICollectionViewFlowLayout and it resolved the issue.

**Comment by mankusti** [comment]
Right on! I wasn't doing anything custom with my collection flow layout but I was recreating it in certain cases. Apparently that wasn't being appreciated by Texture. Now it works. Thanks for the pointer!
